src/main/docker/gitlab-jenkins.yml

```diff
@@ -25,15 +25,15 @@ services:
         hostname: jenkins
         networks:
             artemis:
-                ipv4_address: 172.19.0.3
+                ipv4_address: 172.20.0.5
 
 networks:
     artemis:
         driver: "bridge"
         ipam:
             driver: default
             config:
-                - subnet: 172.19.0.0/24
+                - subnet: 172.20.0.0/24
```

Comment:
Was this change intended?
Now the subnets of the networks defined inside `atlassian.yml` and `gitlab-jenkins.yml` overlap, wheres `atlassian.yml` and `gitlab-jenkins-mysql.yml` do not overlap.

I think we should be consistent here (and maybe also remove `gitlab-jenkins.yml` or restructure it to remove duplication?).

Reply:
These changes actually came from @Hialus' contributions to this PR. But you're right that we should be consistent.

TBH, I'm not sure what's the best solution. On the one hand, having the same subnet for all configs might result in problems, if e.g. someone runs multiple configs (GitLab & Atlassian stack) at the same time (or at least creates the containers with compose). We would have to make sure, that the IP addresses don't collide. The IP `172.xxx.0.2`, e.g., is assigned to multiple services across all configs.

I would suggest that we stick to individual subnets for each service group. This way, they won't collide with each other at all.

Reply:
Regarding the `gitlab-jenkins.yml` file:
I'm not quite sure what this file is intended for as I could not find any references to it in the documentation. But from its structure, it looks as if this file is for rather production-like systems, doesn't it?

If this is the case, we could aim for a "base" file that can be used in production (with CSRF enabled and no pre-set `master.key`) and provide an override file (like we do for the C and Swift images) that then can be used when running a cluster for local development.

But if the file is there for no valid reason, I think we should remove it. If someone doesn't need a DB, e.g., this service could just be commented out...